src/streaming-services/nrk/NrkApi.ts

```diff
@@ -65,23 +129,29 @@ class _NrkApi extends Api {
 			}
 			let isLastPage = false;
 			let items: Item[] = [];
-			const historyItems: NrkHistoryItem[] = [];
+			const historyItems: NrkProgressItem[] = [];
 			do {
 				const responseText = await Requests.send({
-					url: `${this.HISTORY_API_URL}?pg=${nextPage}`, //TODO figure out if pagination is even supported in the API
+					url: this.HISTORY_API_URL,
 					method: 'GET',
+					headers: {
+						Authorization: 'Bearer ' + this.token,
+					},
 				});
-				const responseJson = JSON.parse(responseText) as NrkHistoryItem[];
-				if (responseJson && responseJson.length > 0) {
-					itemsToLoad -= responseJson.length;
-					historyItems.push(...responseJson);
+				const responseJson = JSON.parse(responseText) as NrkProgressResponse;
+				const { progresses } = responseJson;
+				this.HISTORY_API_URL = this.API_HOST_URL + responseJson._links.next.href;
```

Comment:
Since this is an API with navigational features, do you think this is the correct way to achieve navigation? The actual URL is not easy to replicate as it consists of a key-param, which functions as the starting point.

Reply:
If the URL is constantly changing, I think this is ok. It might be preferable to rename it as `historyApiUrl` just to separate the notion of constants and variables, so we know that this value will change, but I don't think it's that important.